### ISSUE TITLE
Update the dependency on reqwest

### ISSUE THREAD

**Issue by ao2** [Aaronepower/Mammut#43]
Hi,

Debian Unstable ships with OpenSSL 1.1.1 and the `reqwest` dependency in Mammut relies on the version 0.9.24 of the openssl crate which does not support OpenSSL 1.1.1.

in particular compilation fails with this message:

```
error: failed to run custom build command for `openssl v0.9.24`
process didn't exit successfully: `/home/ao2/WIP/mastodon/mastodon-twitter-sync/Mammut/target/debug/build/openssl-3a8db21765a1c7e0/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Unable to detect OpenSSL version', /home/ao2/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

Updating to `reqwest = "0.9"` would fix the openssl issue but it looks like the API has changed and the code needs some updates too:

```
    Compiling mammut v0.11.0 (file:///home/ao2/WIP/mastodon/mastodon-twitter-sync/Mammut)
error[E0432]: unresolved imports `reqwest::header::Link`, `reqwest::header::RelationType`
 --> src/page.rs:3:23
  |
3 | use reqwest::header::{Link, RelationType};
  |                       ^^^^  ^^^^^^^^^^^^ no `RelationType` in `header`
  |                       |
  |                       no `Link` in `header`

error[E0432]: unresolved imports `reqwest::header::Authorization`, `reqwest::header::Bearer`, `reqwest::header::Headers`
  --> src/lib.rs:66:23
   |
66 | use reqwest::header::{Authorization, Bearer, Headers};
   |                       ^^^^^^^^^^^^^  ^^^^^^  ^^^^^^^ no `Headers` in `header`
   |                       |              |
   |                       |              no `Bearer` in `header`
   |                       no `Authorization` in `header`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0432`.
error: Could not compile `mammut`.

To learn more, run the command again with --verbose.
```

I'll try looking into it, but feel free to anticipate me :smile:.

Thanks,
   Antonio

BTW the issue arose when compiling https://github.com/klausi/mastodon-twitter-sync

**Comment by Aaronepower** [comment]
Thank you for your issue! Unfortunately the nature of how the API has changed requires a lot more work to update than I'm willing to put in at the moment. If someone wants to pick this up the main pain point is moving `src/page.rs`'s `get_link` function to manually parse out `prev` and `next` links out of the `Link` header (You could probably pull out the code from [hyper 0.11](https://github.com/hyperium/hyper/tree/0.11.x)). This would also probably be fixed by [`http` adding back typed based headers](https://github.com/hyperium/http/issues/264) so that we don't have to manually parse the links ourselves.

**Comment by Aaronepower** [comment]
This has now been fixed with the release of 0.12

**Comment by ao2** [comment]
Thank you @Aaronepower.
